File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/command/solution/TurnRightCommand.java
// TurnRightCommand.java

public class TurnRightCommand extends SimpleDriverCommand
{
	public TurnRightCommand(Driver aDriver)
	{
		super(aDriver);
	}

	public void execute()
	{
		getDriver().turnRight();
	}

	public void undo()
	{
		getDriver().undoTurnRight();
	}
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/command/solution/turn_right_command.cpp
// turn_right_command.cpp

#include "driver.h"
#include "turn_right_command.h"

turn_right_command::turn_right_command(driver * a_driver)
: simple_driver_command(a_driver)
{
}

void turn_right_command::execute()
{
	get_driver()->turn_right();
}

void turn_right_command::undo()
{
	get_driver()->undo_turn_right();
}
